### ISSUE TITLE
Add fine-tuning documentation

### ISSUE THREAD

**Issue by johnbradley** [Imageomics/pybioclip#76]
Provide documentation on how users could fine-tune BioCLIP.

**Comment by johnbradley** [comment]
My understanding of the three example above: These use only [the visual part of the BioCLIP model](https://github.com/samuelstevens/biobench/blob/637432bfda2b567d966d49bf8c4b37b339d4dc2a/biobench/third_party_models.py#L71C22-L71C33) and instead train SVM/ridge classifier/simpleshot based on the image embeddings. No text embeddings are used.

**Comment by samuelstevens** [comment]
Yep, that's correct. biobench is a method to evaluate vision-only models like the vision encoder from BioCLIP, or DINOv2, etc.

**Comment by johnbradley** [comment]
@samuelstevens I want to add some documentation comparing the different approaches above. How would you suggest a user go about deciding between the three options?

**Comment by hlapp** [comment]
Maybe we should try and ask Copilot or ChatGPT, or as seems to the latest rave, the most recent version of DeepSeek 😉 (Though only half joking)

**Comment by samuelstevens** [comment]
Yeah I actually think asking an LLM for a summary is a good idea. Fundamentally, BioCLIP embeddings are designed to be linearly separable. The SVM and ridge classifier are linear (unless you choose a nonlinear kernel in the SVM), while the KNN/Simpleshot is non-linear. 

Another point is that SVM and ridge classifier are parametric ML methods that explicitly learn parameters, while KNN is non-parametric. But in practice, all three will produce similar results, and I'm not sure that I have any recommendations on what methods are likely to produce the best results based on different tasks. 

BioBench would be a good way to test this but it's not set up to evaluate multiple ML methods on a given task right now---I am trying to write a short paper for CV4Animals about BioBench and some findings, but I won't get to that until February. If you're willing to wait, I am happy to share any insights I gain from that project.

**Comment by johnbradley** [comment]
## Asking ChatGPT
```
Compare SVM, Ridge Classifier, and SimpleShot for fine tuning image embeddings 
that linearly separable for classification.
```
## ChatGPT Output

<img width="793" alt="Image" src="https://github.com/user-attachments/assets/dde6144f-df12-4aea-895d-68f85cbb055f" />

### Recommendations
- **SVM**: Best suited for tasks where embeddings are high-quality and you need precise decision boundaries.
- **Ridge Classifier**: Ideal for large datasets where computational efficiency is crucial and embeddings are linearly separable with noise.
- **SimpleShot**: Excellent for few-shot learning scenarios, or when you want a fast, non-parametric approach without retraining.

If your embeddings are linearly separable and classification is the goal, **SVM** often provides the best performance in terms of accuracy, but **Ridge Classifie**r may be preferred for faster computation. If simplicity or few-shot adaptability is key, go for **SimpleShot**.

**Comment by johnbradley** [comment]
[Link to full details from ChatGPT](https://chatgpt.com/share/6797988c-a37c-8005-bf20-d8d70e10a560)

**Comment by hlapp** [comment]
Perhaps @samuelstevens can double check what is true and what might be hallucinations, but I'd say pretty good on the face of it.

**Comment by samuelstevens** [comment]
Perhaps the best strengths of each are:

1. SVMs can support linear and non-linear boundaries and are suitable for binary classification or fewer than ~20 classes (because you train a one-vs-rest for each class).
2. Ridge classifiers are best for linear classification tasks. They require training but are powerful classifiers for many, many tasks, especially with sufficient data.
3. SimpleShot is extremely data-efficient and works well for multiple (20+) classes.

This table summarizes this well I think:

| Method | Maximum Classes | Minimum Training Data | 
|---|---|---|
| SVM | ~20 | 5+ examples |
| Ridge Classifier | No maximum | 10+ examples per class |
| SimpleShot | No maximum | 1+ example per class |

So if you have 3 classes with 5 examples each, I would expect SVMs > SimpleShot > Ridge Classifier. If you have 40 classes with 50 examples each, I would expect Ridge Classifier > SimpleShot, and I would expect SVMs to take too long to train reliably.

Does that clarify things?

**Comment by johnbradley** [comment]
As of #98 I think this has been implemented. Closing.
